Examples/Detectors/DD4hepDetector/src/DD4hepGeometryService.cpp

```diff
@@ -17,9 +17,15 @@
 #include "Acts/Plugins/DD4hep/ConvertDD4hepDetector.hpp"
 #include "Acts/Utilities/BinningType.hpp"
 
+#include <stdexcept>
+
 ActsExamples::DD4hep::DD4hepGeometryService::DD4hepGeometryService(
     const ActsExamples::DD4hep::DD4hepGeometryService::Config& cfg)
-    : BareService("DD4hepGeometryService", cfg.logLevel), m_cfg(cfg) {}
+    : BareService("DD4hepGeometryService", cfg.logLevel), m_cfg(cfg) {
+  if (m_cfg.xmlFileNames.empty()) {
+    throw std::invalid_argument("Missing DD4hep XML filenames");
+  }
```

Comment:
@msmk0 While trying out Acts 4.1, I discovered that this error handling code is not reached in practice. The program bombs at boost::program_options parsing time with...

```
terminate called after throwing an instance of 'boost::wrapexcept<boost::bad_any_cast>'                                                                                                                                                                                             
  what():  boost::bad_any_cast: failed conversion using boost::any_cast
```

...and a quick GDB run points me to `DD4hepDetectorOptions.hpp:103`, i.e. the point where the DD4hep detector options are parsed.

It seems that not specifying a CLI argument is not a valid configuration when asking boost::program_options to parse an `std::vector<std::string>`.

Reply:
Thanks for catching this. 

>It seems that not specifying a CLI argument is not a valid configuration when asking boost::program_options to parse an std::vector<std::string>.

This is a bit strange, because from what I understood that is exactly the kind of usage for these kind of options. Maybe we need to set a default value equivalent to an empty vector to make this work.